azurerm/internal/services/web/app_service.go

```diff
@@ -490,6 +490,15 @@ func schemaAppServiceSiteConfig() *pluginsdk.Schema {
 					Type:     pluginsdk.TypeString,
 					Optional: true,
 				},
+				"acr_use_managed_identity_creds": {
+					Type:     pluginsdk.TypeBool,
+					Optional: true,
+					Default:  false,
+				},
+				"acr_user_managed_identity_id": {
+					Type:     pluginsdk.TypeString,
+					Optional: true,
+				},
 			},
```

Comment:
```suggestion
				},
				
				"acr_use_managed_identity_credentials": {
					Type:     pluginsdk.TypeBool,
					Optional: true,
					Default:  false,
				},
				
				"acr_user_managed_identity_id": {
					Type:     pluginsdk.TypeString,
					Optional: true,
					ValidateFunc: validate.UserAssignedIdentityID,
				},
			},
```
Can we add the validation back in here. Whilst this property can take just the client_id it's more common to use the full resource id to pass between resources, which is also valid here.

Reply:
No, as the SDK only works when I set the client ID. As in ￼will pull a container from an ARC.

Reply:
I tried added a test that would pull the container but I could not complete a local run with null resource and the azure cli to import a container image.

Reply:
@mozts2005 - This feels like a bug in the service, however, if this the case, we should rename this property to match the required name `acr_user_managed_identity_client_id` to be specific about the value and avoid confusion about which ID is required. (since normal parlance means the resource ID).

Reply:
Ok, I will change the name.

Reply:
Change the names as requested